src/test/java/com/amazon/ion/impl/bin/IonManagedWriter_1_1_Test.kt

```diff
@@ -97,4 +143,612 @@ internal class IonManagedWriter_1_1_Test {
             }
         }
     }
+
+    @Test
+    fun `write an encoding directive with a non-empty macro table`() {
+        val expected = """
+            $ion_1_1
+            $ion_encoding::((macro_table (macro null () "foo")))
+        """.trimIndent()
+
+        val actual = write {
+            addMacro(constantMacro { string("foo") })
+        }
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `write an e-expression by name`() {
+        val expected = """
+            $ion_1_1
+            $ion_encoding::((macro_table (macro a () "foo")))
+            (:a)
+        """.trimIndent()
+
+        val actual = write {
+            val macroRef = addMacro("a", constantMacro { string("foo") })
+            startMacro(macroRef)
+            endMacro()
+        }
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `write an e-expression by name 2`() {
```

Comment:
Is this just a duplicate of the previous test?

Reply:
Yes, it is. Oops.